src/components/textfieldSlider/js/TextfieldSlider.js

```diff
@@ -66,123 +62,117 @@ var fluid_2_0 = fluid_2_0 || {};
 
     fluid.defaults("fluid.textfieldSlider.textfield", {
-        gradeNames: ["fluid.viewComponent", "autoInit"],
+        gradeNames: ["fluid.viewRelayComponent", "autoInit"],
+        range: {}, // should be used to specify the min, max range e.g. {min: 0, max: 100}
+        modelRelay: {
+            target: "value",
+            singleTransform: {
+                type: "fluid.transforms.stringToNumber",
+                input: "{that}.model.stringValue"
+            }
+        },
+        modelListeners: {
+            value: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{change}.value"]
+            },
+            // When the input field receives a invalid input such as a string value, ignore it and populate the field with the previous value
+            stringValue: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{that}.model.value"]
+            }
+        },
         listeners: {
-            onCreate: {
-                listener: "fluid.textfieldSlider.textfield.init",
-                args: "{that}"
+            "onCreate.bindChangeEvt": {
+                "this": "{that}.container",
+                "method": "change",
+                "args": ["{that}.setModel"]
             }
         },
-        range: {} // should be used to specify the min, max range e.g. {min: 0, max: 100}
-    });
-
-    fluid.textfieldSlider.validateValue = function (model, range, changeRequest) {
-        var oldValue = model.value;
-        var newValue = changeRequest.value;
-
-        if (!isNaN(parseInt(newValue, 10))) {
-            if (newValue < range.min) {
-                newValue = range.min;
-            } else if (newValue > range.max) {
-                newValue = range.max;
+        invokers: {
+            setModel: {
+                changePath: "stringValue",
+                value: "{arguments}.0.target.value"
             }
-            changeRequest.value = Number(newValue);
-        } else {
-            changeRequest.value = oldValue;
         }
-    };
-
-    fluid.textfieldSlider.textfield.init = function (that) {
-        that.applier.guards.addListener({path: "value", transactional: true}, function (model, changeRequest) {
-            fluid.textfieldSlider.validateValue(model, that.options.range, changeRequest);
-        });
-
-        that.container.change(function (source) {
-            that.applier.requestChange("value", source.target.value);
-        });
-    };
+    });
 
     fluid.defaults("fluid.textfieldSlider.slider", {
-        gradeNames: ["fluid.viewComponent", "autoInit"],
+        gradeNames: ["fluid.viewRelayComponent", "autoInit"],
+        range: {}, // should be used to specify the min, max range e.g. {min: 0, max: 100}
         selectors: {
             thumb: ".ui-slider-handle"
         },
-        events: {
-            modelChanged: null
+        members: {
+            slider: {
```

Comment:
The member "slider" does return a jQuery slider object that has all slider functions attached onto. If I move initSlider to an "onCreate" listener, the instantiation of the slider would happen too late that the initial triggering of the model listener to set the initial slider value (https://github.com/cindyli/infusion/blob/FLUID-5552/src/components/textfieldSlider/js/TextfieldSlider.js#L163) complaints the undefined function of slider().

As a result, the expander for initializing the slider member option is kept. But the code is refactored to get rid of the oddity of passing setSliderAria() as an argument to the invoker setSliderValue().